cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -1040,6 +1057,40 @@ protected Future<Void> updateKafkaVersion(KafkaStatus kafkaStatus) {
         return Future.succeededFuture();
     }
 
+    protected Future<Void> updateKafkaMetadataState(KafkaStatus kafkaStatus) {
+        kafkaStatus.setKafkaMetadataState(kafkaMetadataStateManager.computeNextMetadataState().name());
+        return Future.succeededFuture();
+    }
+
+    protected Future<Void> maybeKafkaMetadataMigrationInProgress() {
+        KafkaMetadataStateManager.KafkaMetadataConfigurationState kafkaMetadataConfigState = this.kafkaMetadataStateManager.computeMetadataConfigurationState();
+        // on each reconcile, would be useless to check migration status if it's not going on
+        if (kafkaMetadataConfigState.equals(KafkaMetadataStateManager.KafkaMetadataConfigurationState.MIGRATION)) {
+            LOGGER.infoCr(reconciliation, "Checking ZooKeeper migration state");
+            return ReconcilerUtils.clientSecrets(reconciliation, secretOperator)
```

Comment:
How many times are we getting the secrets? Aren't they already stored somewhere?

Reply:
In this specific case, we are getting secrets only when there is a "migration" ongoing.
I can't find a place where they are stored tbh. I see that the same way of doing that (so by using `ReconcilerUtils.clientSecrets`) is also done on each reconcile when retrieving the cluster id and during the "maybe" roll Kafka part.
Maybe we could retrieve them one time at the beginning of the reconciliation in order to be used in these three specific steps but ... 
1. not sure if any certificates change could have an impact on using "something" you retrieve at the beginning of the reconcile (even if reconciling certs happens before reconciling Kafka so we should not have such a problem)
2. I would leave this for a different PR because it would have impact on code which is not migration related only